src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2061,7 +2061,7 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
             begin = pos;
         }
 
-        if (begin == 0 && pos == std::wstring::npos)
+        if (begin == 0)

```

Comment:
I'm not certain I agree with that. If you pass in `hello\n`:

```
stripped = hello\r
wstr     = hello\n
begin    = 0
pos      = 5
```

we will erroneously send wstr (`\n`) instead of stripped (`\r`)

Reply:
Wouldn't `begin` be equal to 5 because of line 2061?

Also since this is after the while loop, `pos` is guaranteed to be `npos` right? Otherwise we would not have exited the loop

Reply:
oh great catch

Reply:
does it deserve a comment explaining how we know we're safe here?

Reply:
Okay will do!

Reply:
Done